cdc/sink/codec/canal.go

```diff
@@ -133,38 +134,47 @@ func (b *canalEntryBuilder) buildHeader(commitTs uint64, schema string, table st
 	return h
 }
 
-func getJavaSQLType(c *model.Column, mysqlType string) (result JavaSQLType) {
+func getJavaSQLType(c *model.Column, mysqlType string) (result JavaSQLType, err error) {
 	javaType := mySQLType2JavaType(c.Type, c.Flag.IsBinary())
 
 	switch javaType {
 	case JavaSQLTypeBINARY, JavaSQLTypeVARBINARY, JavaSQLTypeLONGVARBINARY:
 		if strings.Contains(mysqlType, "text") {
-			return JavaSQLTypeCLOB
+			return JavaSQLTypeCLOB, nil
 		}
-		return JavaSQLTypeBLOB
+		return JavaSQLTypeBLOB, nil
 	}
 
 	// flag `isUnsigned` only for `numerical` and `bit`, `year` data type.
 	if !c.Flag.IsUnsigned() {
-		return javaType
+		return javaType, nil
 	}
 
 	// for year, to `int64`, others to `uint64`.
 	// no need to promote type for `year` and `bit`
 	if c.Type == mysql.TypeYear || c.Type == mysql.TypeBit {
-		return javaType
+		return javaType, nil
 	}
 
 	if c.Type == mysql.TypeFloat || c.Type == mysql.TypeDouble || c.Type == mysql.TypeNewDecimal {
-		return javaType
+		return javaType, nil
 	}
 
-	// for **unsigned** integral types, should have type in `uint64`. see reference:
-	// https://github.com/pingcap/ticdc/blob/f0a38a7aaf9f3b11a4d807da275b567642733f58/cdc/entry/mounter.go#L493
+	// for **unsigned** integral types, type would be `uint64` or `string`. see reference:
+	// https://github.com/pingcap/tiflow/blob/1e3dd155049417e3fd7bf9b0a0c7b08723b33791/cdc/entry/mounter.go#L501
 	// https://github.com/pingcap/tidb/blob/6495a5a116a016a3e077d181b8c8ad81f76ac31b/types/datum.go#L423-L455
-	number, ok := c.Value.(uint64)
-	if !ok {
-		log.Panic("unsigned value not in type uint64", zap.Any("column", c))
+	var number uint64
+	switch v := c.Value.(type) {
+	case uint64:
+		number = v
+	case string:
+		a, err := strconv.ParseUint(v, 10, 64)
+		if err != nil {
+			return javaType, err
+		}
+		number = a
+	default:
+		return javaType, errors.Errorf("unexpected type for unsigned value: %+v", reflect.TypeOf(v))
```

Comment:
Pls don't remove zap.Any("column", c), it will offer detail infomation.
Others LGTM.